src/Aspire.Hosting.Kubernetes/KubernetesResource.cs

```diff
@@ -92,8 +94,41 @@ internal IEnumerable<BaseKubernetesResource> GetTemplatedResources()
         }
     }
 
-    private void BuildKubernetesResources()
+    private void BuildKubernetesResources(KubernetesEnvironmentContext context, DistributedApplicationExecutionContext executionContext)
     {
+        // Process deferred environment variables
+        foreach (var environmentVariable in RawEnvironmentVariables)
```

Comment:
Move the RawEnvironmentVariables and RawArguments processing should move to a separate method and be called from here.